numba/tests/test_serialize.py

```diff
@@ -12,11 +12,18 @@
 from numba.tests.support import TestCase
 from numba.core.target_extension import resolve_dispatcher_from_str
 from numba.cloudpickle import dumps, loads
-from .serialize_usecases import *
 
 
 class TestDispatcherPickling(TestCase):
 
+    def setUp(self):
+        super().setUp()
+        # delay import of these so as to not trigger e.g. deprecation warnings
+        # present on Numba decorator APIs on import.
+        import numba.tests.serialize_usecases as ucs
+        g = globals()
+        g.update({x: getattr(ucs, x) for x in ucs.__all__})
```

Comment:
shouldn't it be a `setUpClass`? or is that too early? Every test method will trigger a global update.

and wouldn't it be better to turn all the references to `serialize_usecase.xyz` instead of hacking the globals.

Reply:
I wasn't sure if the original import form had a specific intent (it's very unusual to see `from .<module> import *` in the code base) and given this is testing serialisation thought it best to attempt to emulate that behaviour. If it's not important, I think it should be ok to put the `serialize_usecases` module as a reference into the test class itself and then refer to it as `self._serialize_usecases.xyz` as noted. What do you think?